job/server/src/main/java/alluxio/job/util/JobUtils.java

```diff
@@ -147,7 +147,7 @@ public static void loadBlock(URIStatus status, FileSystemContext context, long b
     // when the data to load is persisted, simply use local worker to load
     // from ufs (e.g. distributed load) or from a remote worker (e.g. setReplication)
     // Only use this read local first method to load if nearest worker is clear
-    if (netAddress.size() <= 1 && pinnedLocation.isEmpty() && status.isPersisted()) {
+    if (netAddress.size() <= 1 && !status.getFileInfo().isPinned() && status.isPersisted()) {
```

Comment:
could you explain why pinned files can not go through this code path? and why this would fix the previous issue? it is not too clear to me.

Reply:
discussed offlline with @jja725 . LGTM